custom/ewsghana/reports/specific_reports/stock_status_report.py

```diff
@@ -334,40 +303,28 @@ def headers(self):
     def rows(self):
         rows = []
         if self.config['location_id']:
-            location = SQLLocation.objects.get(
-                domain=self.config['domain'],
-                location_id=self.config['location_id']
-            )
-            if location.location_type.name == 'country':
-                supply_points = SQLLocation.objects.filter(
-                    Q(parent__location_id=self.config['location_id']) |
-                    Q(location_type__name='Regional Medical Store', domain=self.config['domain'])
-                ).order_by('name').exclude(supply_point_id__isnull=True).exclude(is_archived=True)
-            else:
-                supply_points = SQLLocation.objects.filter(
-                    parent__location_id=self.config['location_id']
-                ).order_by('name').exclude(supply_point_id__isnull=True).exclude(is_archived=True)
-
-            products = set(self.unique_products(supply_points))
-            for supply_point in supply_points:
-                product_map = {p.product_id: p.name for p in products.intersection(set(supply_point.products))}
-                stockout = StockTransaction.objects.filter(
-                    type='stockonhand',
-                    product_id__in=product_map.keys(),
-                    case_id=supply_point.supply_point_id,
-                    report__date__range=[self.config['startdate'], self.config['enddate']],
-                    stock_on_hand=0
-                ).order_by('product_id', '-report__date').distinct('product_id').values_list('product_id',
-                                                                                             flat=True)
-                if stockout:
-                    url = make_url(
-                        StockLevelsReport,
-                        self.config['domain'],
-                        '?location_id=%s&startdate=%s&enddate=%s',
-                        (supply_point.location_id, self.config['startdate'], self.config['enddate'])
+            product_id_to_name = {
+                product_id: product_name
+                for (product_id, product_name) in self.config['unique_products'].values_list('product_id', 'name')
+            }
+            for supply_point_id in self.config['stockout_table_supply_points']:
+                supply_point = SQLLocation.objects.get(supply_point_id=supply_point_id)
```

Comment:
If `self.config['stockout_table_supply_points']` is being used in multiple places, I imagine each of those might want access to the `SQLLocation` object.

Also, this performs `len(supply_point_ids)` database calls
```python
for supply_point_id in supply_point_ids:
    supply_point = SQLLocation.objects.get(supply_point_id=supply_point_id)
    ...
```
Whereas this only performs one:
```python
for supply_point in SQLLocation.objects.filter(supply_point_id__in=supply_point_ids):
    ...
```
What I'm getting at is that this should probably move the database calls back to the initialization of `self.config`, which should return the full objects, if they're needed.